libevmasm/RuleList.h

```diff
@@ -597,6 +626,24 @@ std::vector<SimplificationRule<Pattern>> simplificationRuleListPart7(
 		}
 	});
 
+	rules.push_back({
+		Builtins::SHL(A, Builtins::SIGNEXTEND(B, X)),
+		[=]() -> Pattern { return Builtins::SIGNEXTEND((A.d() >> 3) + B.d(), Builtins::SHL(A, X)); },
+		[=] { return (A.d() & 7) == 0 && A.d() <= Pattern::WordSize && B.d() <= Pattern::WordSize / 8; }
+	});
+
+	rules.push_back({
+		Builtins::SIGNEXTEND(A, Builtins::SHR(B, X)),
+		[=]() -> Pattern { return Builtins::SAR(B, X); },
+		[=] {
+			return
+				B.d() % 8 == 0 &&
+				B.d() <= Pattern::WordSize &&
+				A.d() <= Pattern::WordSize &&
+				(Pattern::WordSize - B.d()) / 8 == A.d() + 1;
+		}
+	});
+
```

Comment:
I've not completely hand checked this. But tried to fuzz with dapptools this myself and seems okay. And the proof for this also seems okay.

The rest of them I checked by hand. 

So approving this.